simulation_test.go

```diff
@@ -167,3 +167,55 @@ func TestFromITF(t *testing.T) {
 
 	}
 }
+
+func TestFromScenario(t *testing.T) {
+	itfFileName := "formal_spec/ITF_files/scenario.json"
+	data, err := ioutil.ReadFile(itfFileName)
+	if err != nil {
+		t.Errorf("Error opening file: %v", err)
+	}
+	tests := gjson.GetBytes(data, "testCases").Array()
+
+	for _, tt := range tests {
+		states := tt.Get("states").Array()
+		lastState := states[len(states)-1]
+		namespaceId := int(lastState.Get("namespace_v").Int())
+
+		itf_leaves := lastState.Get("leaves_namespace_idx_v").Array()
+
+		tree := nmt.New(sha256.New(), nmt.NamespaceIDSize(nidSize))
+		for _, itf_leaf := range itf_leaves {
+			leaf_id := int(itf_leaf.Int())
+			pushData := namespace.PrefixedData(append(
+				intToBytes(leaf_id, nidSize),
+				[]byte("dummy_data")...))
+			err := tree.Push(pushData)
+			if err != nil {
+				t.Fatalf("invalid test case: %v, error on Push(): %v", tt.Get("name").String(), err)
+			}
+		}
+		proveNID := intToBytes(namespaceId, nidSize)
+		gotProof, err := tree.ProveNamespace(proveNID)
+		if err != nil {
+			t.Fatalf("ProveNamespace() unexpected error: %v", err)
+		}
+
+		gotFound := gotProof.IsNonEmptyRange() && len(gotProof.LeafHash()) == 0
+		wantFound := lastState.Get("namespaceIdFound_v").Bool()
+
+		if gotFound != wantFound {
+			t.Errorf("Proof.ProveNamespace() gotFound = %v, wantFound = %v ", gotFound, wantFound)
+		}
+
+		// Verification round-trip should always pass:
+		gotGetLeaves := tree.Get(proveNID)
```

Comment:
Thanks @ivan-gavran for providing the sample for generating unit test data. 
At this line, I noticed that it doesn't appear to correspond to any section in the original `TestNamespacedMerkleTree_ProveNamespace_Ranges_And_Verif` test from `nmt_test.go`. Can you please confirm if my understanding is correct?